src/components/questionnaire/questionnaire-interaction-screen.tsx

```diff
@@ -0,0 +1,160 @@
+import { KeyboardReturn } from "@mui/icons-material";
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Checkbox,
+  Divider,
+  FormControlLabel,
+  Typography
+} from "@mui/material";
+import { useState } from "react";
+import type { AnswerOption, Question, Questionnaire } from "src/generated/homeLambdasClient";
+import strings from "src/localization/strings";
+
+interface QuestionnaireInteractionScreenProps {
+  questionnaire: Questionnaire | null;
+  mode: "view" | "edit" | "preview";
+  onBack: () => void;
+}
+
+/**
+ *  Screen for the user to interact with the questionnaire; fill, edit and preview
+ */
+const QuestionnaireInteractionScreen: React.FC<QuestionnaireInteractionScreenProps> = ({
+  questionnaire,
+  mode,
+  onBack
+}) => {
+  if (!questionnaire) {
+    return <Typography variant="h4">Loading...</Typography>;
+  }
+
+  const [responses, setResponses] = useState<Record<string, string[]>>({});
+
+  const handleOptionChange = (questionId: string, optionLabel: string, isChecked: boolean) => {
+    setResponses((prevResponses) => {
+      const selectedOptions = prevResponses[questionId] || [];
+      return {
+        ...prevResponses,
+        [questionId]: isChecked
+          ? [...selectedOptions, optionLabel]
+          : selectedOptions.filter((label) => label !== optionLabel)
+      };
+    });
+  };
+
+  const handleSubmit = () => {
+    let correctAnswersCount = 0;
+    questionnaire.questions.forEach((question) => {
+      const userAnswers = responses[question.questionText] || [];
```

Comment:
I'm wondering about the use of questionText as a stand in for questionId. I think  in this context it's fine and questions don't actually need an id. But have a think about it also, we could update the specs if needed.